dashboard/app/models/user.rb

```diff
@@ -2133,6 +2144,11 @@ def self.track_level_progress(user_id:, level_id:, script_id:, new_result:, subm
       total_time_spent = user_level.calculate_total_time_spent(time_spent)
       user_level.time_spent = total_time_spent if total_time_spent
 
+      if locale
+        user_level.locale = locale
+        user_level.locale_supported = script.supported_locale?(locale)
```

Comment:
@code-dot-org/red-team Please let me know if you have a better name for the flag (e.g. as `supports_locale`)